src/switch_core_media.c

```diff
@@ -8830,12 +8830,18 @@ SWITCH_DECLARE(switch_status_t) switch_core_session_start_video_thread(switch_co
 	return SWITCH_STATUS_SUCCESS;
 }
 
-SWITCH_DECLARE(void) switch_core_media_start_engine_function(switch_core_session_t *session, switch_media_type_t type, switch_engine_function_t engine_function, void *user_data)
+SWITCH_DECLARE(void) switch_core_media_start_engine_function(const char *uuid, switch_media_type_t type, switch_engine_function_t engine_function, void *user_data)
```

Comment:
I don't see a point of passing the uuid here since this should be called in the same thread. So session should be valid or lock until the end of this function. I would advice that you still get the session as a parameter to minimize the change and only pass the UUID in the user data upon starting the video thread.